service/history/timerQueueProcessorBase.go

```diff
@@ -238,8 +247,7 @@ func (t *timerQueueProcessorBase) internalProcessor() error {
 		//
 		select {
 		case <-t.shutdownCh:
-			t.logger.Debug("Timer queue processor pump shutting down.")
-			return nil
+			break eventLoop
```

Comment:
return nil?

Reply:
nil is returned at the very end of this function